File: mvc-java/src/main/java/org/ivan/config/InitializerWebApp.java
package org.ivan.config;

import org.springframework.web.servlet.resource.ResourceUrlEncodingFilter;
import org.springframework.web.servlet.support.AbstractAnnotationConfigDispatcherServletInitializer;

import javax.servlet.Filter;

public class InitializerWebApp extends AbstractAnnotationConfigDispatcherServletInitializer {
    @Override protected Class<?>[] getRootConfigClasses() {
        return new Class[] { ConfigApp.class };
    }

    @Override protected Class<?>[] getServletConfigClasses() {
        return new Class[] { ConfigWeb.class };
    }

    @Override protected String[] getServletMappings() {
        return new String[] { "/" };
    }

    @Override
    protected Filter[] getServletFilters() {
        return new Filter[] {
                new ResourceUrlEncodingFilter()
//            new HiddenHttpMethodFilter(),
//            new CharacterEncodingFilter()
        };
    }
}

File: mvc-kotlin/src/main/kotlin/org/ivan/config/InitializerWebApp.kt
package org.ivan.config

import org.springframework.web.servlet.resource.ResourceUrlEncodingFilter
import org.springframework.web.servlet.support.AbstractAnnotationConfigDispatcherServletInitializer
import javax.servlet.Filter

class InitializerWebApp : AbstractAnnotationConfigDispatcherServletInitializer() {
    override fun getRootConfigClasses(): Array<Class<*>>? = arrayOf(ConfigApp::class.java);

    override fun getServletConfigClasses(): Array<Class<*>>? = arrayOf(ConfigWeb::class.java);

    override fun getServletMappings(): Array<String> = arrayOf("/")

    override fun getServletFilters(): Array<Filter>? = arrayOf(
            ResourceUrlEncodingFilter()
    )
}